### PR TITLE
R2 make interference

### LINKED ISSUE #39: make-interference
**TODO:**
- [ ] fix the bug below (probably by handling multiple blocks in the main program function)
- [ ] add any new match-arms to handle new clauses in x86_1 grammar

#

The following errors were found after 451c6be
```bash
> ./run programs/if-should-exit-42.rc
Bad input to make interference
...
```

```
> ./run programs/test-spill-regs-should-exit-18.rc
Bad input to make interference
...
```

### PR DESCRIPTION
**Pull request by mfekadu**
resolves #39 ??

### CODE REVIEW

File: testing/allocate-registers-tests.rkt [hunk 1]
```diff
@@ -67,7 +67,7 @@
 (define given1-locals '(t.1 z w y x v))
 
 (define given1-alloc-block `(block () ,given1-alloc-instrs))
-(define given1-alloc-prog `(program ((locals ,given1-locals) (conflicts ,given1-cg)) ,given1-alloc-block))
+(define given1-alloc-prog `(program ((locals ,given1-locals) (conflicts ,given1-cg)) (,given1-alloc-block)))
```
Comment (mfekadu) [inline testing/allocate-registers-tests.rkt]: http://www.commitlogsfromlastnight.com 🤣

**Comment by johnvtan** [comment]
```racket
((label block2861) (movq (int 42) (reg rax)) (jmp conclusion))
((label block2862) (movq (int 44) (reg rax)) (jmp conclusion))
((label block2863) (cmpq (var complex-if2860) (int 1)) (jmp-if e block2861) (jmp block2862))
((label block2864) (movq (int 0) (var complex-if2860)) (jmp block2863))
((label block2865) (cmpq (int 2) (int 1)) (set e (byte-reg al)) (movzbq (byte-reg al) (var complex-op2859)) (cmpq (int 0) (var complex-op2859)) (set e (byte-reg al)) (movzbq (byte-reg al) (var complex-if2860)) (jmp block2863))
((label start) (cmpq (int 2) (int 1)) (set l (byte-reg al)) (movzbq (byte-reg al) (var complex-op2858)) (cmpq (var complex-op2858) (int 1)) (jmp-if e block2864) (jmp block2865))
```

**Comment by johnvtan** [comment]
```racket
(program (complex-op2858 complex-if2860 complex-op2859)

((block (#<set:> #<set:> #<set:> #<set:>) 
(label block2861) (movq (int 42) (reg rax)) (jmp conclusion)) 

(block (#<set:> #<set:> #<set:> #<set:>)
(label block2862) (movq (int 44) (reg rax)) (jmp conclusion))

(block (#<set: complex-if2860> #<set: complex-if2860> #<set:> #<set:> #<set:>) 
(label block2863) (cmpq (var complex-if2860) (int 1)) (jmp-if e block2861) (jmp block2862)) 

(block (#<set:> #<set:> #<set: complex-if2860> #<set: complex-if2860>)
(label block2864) (movq (int 0) (var complex-if2860)) (jmp block2863)) 

(block (#<set:> #<set:> #<set:> #<set:> #<set:> #<set:> #<set:> #<set: complex-if2860> #<set: complex-if2860>) 
(label block2865) (cmpq (int 2) (int 1)) (set e (byte-reg al)) (movzbq (byte-reg al) (var complex-op2859)) (cmpq (int 0) (var complex-op2859)) (set e (byte-reg al)) (movzbq (byte-reg al) (var complex-if2860)) (jmp block2863)) 

(block (#<set:> #<set:> #<set:> #<set:> #<set: complex-op2858> #<set:> #<set:> #<set:>) 
(label start) (cmpq (int 2) (int 1)) (set l (byte-reg al)) (movzbq (byte-reg al) (var complex-op2858)) (cmpq (var complex-op2858) (int 1)) (jmp-if e block2864) (jmp block2865))))
```

**Comment by johnvtan** [comment]
```racket
(program (x2858 complex-op2859) 
((block (#<set:> #<set:> #<set: x2858> #<set: complex-op2859> #<set: complex-op2859> #<set:> #<set:>) 
(label start) (movq (int 2) (var x2858)) (movq (var x2858) (var complex-op2859)) (addq (int 2) (var complex-op2859)) (movq (var complex-op2859) (reg rax)) (jmp conclusion))))
```

**Pull request merged**